clients/amqp-common/src/main/java/org/eclipse/hono/client/amqp/AbstractServiceClient.java

```diff
@@ -264,13 +269,29 @@ public void registerLivenessChecks(final HealthCheckHandler livenessHandler) {
         // no liveness checks to be added
     }
 
+    /**
+     * Adds a handler to be invoked with a succeeded future once this client is ready to be used.
+     *
+     * @param handler The handler to invoke. The handler will never be invoked with a failed future.
+     */
+    public final void addOnClientReadyHandler(final Handler<AsyncResult<Void>> handler) {
+        if (handler != null) {
+            lifecycleStatus.addOnStartedHandler(handler);
+        }
+    }
+
     /**
      * {@inheritDoc}
      *
      * @return The outcome of the connection's {@link HonoConnection#connect()} method.
      */
     @Override
     public Future<Void> start() {
+        if (lifecycleStatus.isStarting()) {
+            return Future.succeededFuture();
+        } else if (!lifecycleStatus.setStarting()) {
+            return Future.failedFuture(new IllegalStateException("sender is already started/stopping"));
+        }
```

Comment:
I take it that subclasses are responsible for invoking *LifecycleStatus.setStarted()*?
If so, we need to document this in the context of this method FMPOV.

Reply:
Oh, yes. This PR is incomplete in that the `lifecycleStatus.setStarted()` invocations are still missing. That involves some more changes.
In any case, having the `addOnClientReadyHandler` method available in the `KafkaApplicationClient` interface is the prime objective here, so I've added a smaller PR (#3282) just with that change.

Reply:
`AbstractServiceClient` changes reverted - see below.